### PR TITLE
Messaging system

### PR DESCRIPTION
**Pull request by victor-paumier**
- Create buddies channels retroactively
- Create buddy channel on match confirmed
- Remove channel on match removed

### CODE REVIEW

File: src/Services/MessagingHelper.php [hunk 1]
```diff
@@ -112,14 +143,31 @@ public function addToThread(
                 $thread->setIsDeletedByParticipant($users, false);
             }
         } else {
-            $thread->addParticipants($users);
+            foreach ($users as $user) {
+                if (null == $thread->getMetadataForParticipant($user)) {
```
Comment (JeremieSamson) [inline src/Services/MessagingHelper.php]:
```suggestion
                if ($thread->getMetadataForParticipant($user)) {
```

File: src/Services/MessagingHelper.php [hunk 2]
```diff
@@ -157,4 +207,9 @@ public function getBCSubject(): string
     {
         return 'CHANNEL_BCS';
     }
+
+    public function getBuddySubject(Buddy $buddy)
+    {
+        return 'BUDDY_'.$buddy->getId();
```
Comment (JeremieSamson) [inline src/Services/MessagingHelper.php]: Shouldn't we put `BUDDY` as a const here ?

File: templates/Message/thread.html.twig [hunk 3]
```diff
@@ -1,10 +1,6 @@
 <div class="thread">
     <div class="thread__header">
-        {% if thread.subject == 'Default' %}
-            {% for participant in thread.participants if participant != app.user %}
-                <p class="thread__user">{{ participant.username }}</p>
-            {% endfor %}
-        {% else %}
+        {% if 'CHANNEL_' in thread.subject %}
```
Comment (JeremieSamson) [inline templates/Message/thread.html.twig]: Shouldn't we put `CHANNEL` as a const here ?

File: templates/Message/thread_header.html.twig [hunk 4]
```diff
@@ -1,20 +1,7 @@
...
-                        'size': '50'
-                    } %}
-                </div>
-                <div class="thread-selector__user hidden-xs">
-                    <span class="thread-selector__name">{{ participant.username }}</span>
-                    <span class="thread-selector__date">{{ thread.lastMessage.createdAt|localizeddate('medium', 'short') }}</span>
-                </div>
-            {% endfor %}
-        {% else %}
+        {% if 'CHANNEL_' in thread.subject %}
```
Comment (JeremieSamson) [inline templates/Message/thread_header.html.twig]: Shouldn't we put `CHANNEL` as a const here ?

Comment (victor-paumier) [reply templates/Message/thread_header.html.twig]: done

File: templates/Message/thread.html.twig (hunk 3 above)
Comment (victor-paumier) [reply templates/Message/thread.html.twig]: done

File: src/Services/MessagingHelper.php (hunk 2 above)
Comment (victor-paumier) [reply src/Services/MessagingHelper.php]: done

File: src/Services/MessagingHelper.php (hunk 1 above)
Comment (victor-paumier) [reply src/Services/MessagingHelper.php]: done

**Pull request merged**